src/main/java/hexgui/sgf/SgfReader.java

```diff
@@ -0,0 +1,411 @@
+      switch (ttype) {
+        case '(':
+          m_tokenizer.pushBack();
+          parseGameTree(node, false);
+          break;
+
+        case ';':
+          m_tokenizer.pushBack();
+          parseNode(node, false);
+          done = true;
+          break;
+
+        case ')':
+          m_tokenizer.pushBack();
+          done = true;
+          break;
+
+        case StreamTokenizer.TT_WORD:
+          parseProperty(node, isroot);
+          break;
+
+        case StreamTokenizer.TT_EOF:
+          throw sgfError("Unexpected EOF in node!");
+
+        default:
+          throw sgfError("Error in SGF file.");
+      }
+    }
+
+    return node;
+  }
+
+  /**
+   * Parse a point or move value. Supports both standard SGF notation for Hex (a1, ...) and Go-like
+   * notation used by Little Golem (aa, ...)
+   */
+  private HexPoint parsePoint(String s) throws SgfError {
+    s = s.trim().toLowerCase(Locale.ENGLISH);
+    HexPoint result = null;
+    if (s.length() >= 2) {
+      if (s.length() == 2 && s.charAt(1) >= 'a' && s.charAt(1) <= 'z') {
+        // Go-like SGF notation
+        int x = s.charAt(0) - 'a';
+        int y = s.charAt(1) - 'a';
+        if (x < HexPoint.MAX_WIDTH && y < HexPoint.MAX_HEIGHT) result = HexPoint.get(x, y);
+      } else {
+        // Standard SGF notation for Hex
+        try {
+          int x = s.charAt(0) - 'a';
+          int y = Integer.parseInt(s.substring(1)) - 1;
+          if (y >= 0 && y < HexPoint.MAX_HEIGHT) result = HexPoint.get(x, y);
+        } catch (NumberFormatException e) {
+        }
+      }
+    }
+    if (result == null) throw sgfError(format("Invalid point {0}", s));
+    return result;
+  }
+
+  private HexPoint parseMove(String s) throws SgfError {
+    s = s.trim().toLowerCase(Locale.ENGLISH);
+
+    // Special case: some or all versions of HexGui up to 0.9.GIT
+    // incorrectly used "swap-pieces" instead of "swap-sides".
+    // The SGF specification states:
+    //
+    // * swap-sides - the player elects to swap sides with his
+    //   opponent; if he was playing Black he now plays White, and
+    //   vice versa.
+    //
+    // * swap-pieces - the player elects to swap pieces with his
+    //   opponent - all of Black's pieces are colored White, and
+    //   White's pieces are colored Black. Then the entire board
+    //   is reflected in the long diagonal axis.
+    //
+    // For backward compatibility, we must compensate for the
+    // incorrect use of "swap-pieces" when reading SGF files
+    // written by HexGui 0.9.GIT or earlier.
+
+    if (m_swap_bug && s.equals("swap-pieces")) {
+      s = "swap-sides";
+    }
+
+    // HexPoint.get() handles special move values like "swap"
+    HexPoint result = HexPoint.get(s);
+    if (result == null)
+      // Handles Go-style point notation (aa, ...)
+      result = parsePoint(s);
+    return result;
+  }
+
+  private void parseProperty(Node node, boolean isroot) throws SgfError, IOException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 23. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.